Added traceback if the urlretrive fails, folder naming, topic download support and fixed doube download bug

szero opened the pull request:
# Here's a link to my fork of this program: https://github.com/Szero/imgur-album-downloader

Added error feedback if the urlretrive fails to download requested in #24

szero committed:
added traceback if the urlretrive fails

nodiscc commented on imguralbum.py:
```diff
@@ -158,7 +159,7 @@ def save_images(self, foldername=False):
                     urllib.request.urlretrieve(image_url, path)
                 except:
-                    print ("Download failed.")
+                    print ("Download failed.\n"+format_exc())
```
Shouldn't it be `traceback.print_exc()`? (https://docs.python.org/3/library/traceback.html)

szero committed:
Merge pull request #1 from alexgisby/master

updating from original

szero committed:
added folder naming, removed unused protocol variable

Creating directories with the same name as the album is a bit problematic because album name may contain characters that are restricted by OS and naming folders with them would cause errors. I overcome this problem by replacing any special characters with `space`

szero commented:
Added folder naming requested in #27 in commit [ 71744f1]

szero committed:
added downloading from topic and fixed double download bug

All because imgur source changed again!

szero commented:
Added downloading from topics requested in #30 in commit [7c52651]

szero commented:
Fixed double download bug requested in #31 in commit [7c52651]

szero committed:
Revert "added downloading from topic and fixed double download bug"

This reverts commit 7c526518db46b9bf69fc6e7f1cd7bc9ea2b95592.

szero committed:
revert to fix conflicts because of editing wrong file

szero committed:
fixed order

szero committed:
typos

szero committed:
made better regex for finding album name

Now program will save files into folders named after album names. If this is undesired folders can be named by their ID like this:

args = sys.argv
downloader = ImgurAlbumDownloader(args[1])
downloader.save_images(downloader.album_key)

cyian-1756 commented:
It errors out with 

`File "imguralbum.py", line 200, in <module>
    downloader = ImgurAlbumDownloader(args[1])
  File "imguralbum.py", line 83, in __init__
    self.album_title = re.search('image\s*:\s*{"id":"%s","title":"(.*?)"' % self.album_key ,html).group(1)
AttributeError: 'NoneType' object has no attribute 'group'`

When downloading _NSFW_[This](https://imgur.com/a/BUfpQ)_NFSW_

szero committed:
fixed attribute error

Another pesky imgur front-end update!

szero committed:
escaping dashes in regex

szero commented:
@cyian-1756 Hey, thanks for the insight. It should be all good now.

cyian-1756 commented:
@Szero Your version can run into issues if it can't find the album name, you might want to check out falling back to the old version naming (The album ID) in that case.

`Traceback (most recent call last):
  File "/home/user/imgur-album-downloader", line 226, in <module>
    downloader.save_images(albumFolder)
  File "/home/user/imgur-album-downloader", line 158, in save_images
    os.makedirs(albumFolder)
  File "/usr/lib/python3.5/os.py", line 241, in makedirs
    mkdir(name, mode)
FileNotFoundError: [Errno 2] No such file or directory: ''`

szero commented:
Could you include link to the image that caused no album match?

On 31/07/16 04:08, cyian-1756 wrote:

> @Szero https://github.com/Szero Your version can run into issues if
> it can't find the album name, you might want to check out falling back
> to the old version naming (The album ID) in that case.
> 
> |Traceback (most recent call last):
> File "/home/user/imgur-album-downloader", line 226, in <module>
> downloader.save_images(albumFolder)
> File "/home/user/imgur-album-downloader", line 158, in save_images
> os.makedirs(albumFolder)
> File "/usr/lib/python3.5/os.py", line 241, in makedirs
> mkdir(name, mode)
> FileNotFoundError: [Errno 2] No such file or directory: ''|
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/alexgisby/imgur-album-downloader/pull/26#issuecomment-236403329,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/APBPc7YByO12Cus_KVTrhjkTKRkslQSHks5qbAONgaJpZM4IDNl1.

cyian-1756 commented:
@Szero It was NSFW [This](https://imgur.com/a/rQ0sK#0) NSFW.

szero committed:
added key album naming when album has no name

szero committed:
added key album naming when album has no name

szero commented:
@cyian-1756 Thanks for the input. I adjusted this for your suggestions.

szero committed:
updated regex for album title matching

szero committed:
removed escape characters from album title string

szero committed:
Revert "removed escape characters from album title string"

This reverts commit 9e3a3052e46ed282ecd02fc8b4bafd5a6a24e41c.

szero committed:
removed escape characters from album title string

cyian-1756 commented:
@Szero I've noticed that if 2 albums have the same name this will put all the downloaded pictures from both albums into one folder, wouldn't it be better (at least for archivers) to add the album ID to the folder if a folder with the same name already exists?

szero committed:
added option to name download folder after album id

szero commented:
@cyian-1756 I added option that names downloaded albums after their ID's. just use it like that: `./imguralbum.py [URL] --use-ids` It will name albums after their ID. Making it automated like you suggests it's not good because if album names are the same that usually means they are the same albums on imgur and it would result in downloading same album twice. Cheers.

szero committed:
fix image hash fetching

ghost commented:
Hi, you pulled me over from the other thread on imgur album downloader. How does this one fix urls not working?

I tried in terminal "python3  [imgur-album-downloader-master in the right location.py] https://imgur.com/gallery/wfMp9 ~/desktop" 

and it told me something similar/the-same-as-the-original-version-of-this-program 

"response_code = e.code
AttributeError: 'URLError' object has no attribute 'code' "

szero committed:
Updates to imguralbum.py

    - Better error handling on urlopen method
    - Added property decorators to 4 methods
    - Refactored some code according to pylint

szero commented:
Hello again, @Coaggly. The stack trace indicates that it is some sort of URL error that wasn't handled properly before. I updated the code so it has more verbose output now. It can be caused by outdated certificates on your system, or something trivial like lack of internet connection. If you are using some Linux-based system, try updating the package lists and upgrade your packages. The package you would be looking to upgrade if it would be the case of the issue is `ca-certificates` on most distros.

ghost commented:
@Szero I'm on mac. I tried updating, same thing. I'm not using homebrew if that makes a difference? I'm really new to this stuff so i don't know if i should be

szero commented:
Are you sure you are running my version? You shouldn't get the same error this time. Please upload full error trace and your python3 version if possible.

ghost commented:
I must apologize, but [google](https://www.google.ca/search?client=opera&hs=uY&ei=bQ8OWoWOKuWUjwSc86GICg&q=error+trace+python&oq=error+trace+python&gs_l=psy-ab.3..0i22i30k1.4138.4942.0.5256.7.6.0.0.0.0.144.398.0j3.3.0....0...1.1.64.psy-ab..4.3.396...0.0.pUyhg-mYHZs) isn't telling me how to get the error trace.

I don't suppose you know a good basic tutorial for python that includes something about trace errors? The ones i watched didn't mention much about error trace.

"Python 3.6.3"

szero commented:
By error trace I just meant what program prints after you run it, it can't be same error this time. Please download my release from here: https://github.com/Szero/imgur-album-downloader/archive/master.zip and try again.
